scripts/metrics/common/query_metrics.py

```diff
@@ -0,0 +1,157 @@
+                    if not os.path.exists(week_folder):
+                        os.mkdir(week_folder)
+                    download_path = os.path.join(download_folder, str(week_number), file_name)
+                    DOWNLOADED_FILES.append(download_path)
+                    bucket.download_file(obj.key, download_path)
+
+            # If a .csv is in the wrong location, it'll break the functionality above. We're just going to skip it.
+            except ValueError as value_error:
+                print("Found value error. Likely means a csv is in the wrong location. Skipping and continuing. Error "
+                      "{0} ".format(str(value_error)))
+
+
+def query_csvs(test_case_ids=None, test_case_metrics=None):
+    """
+    Queries the desired information from the downloaded csvs
+    :param test_case_ids: the desired test case ids
+    :param test_case_metrics: the desired test case metrics
+    """
+    query_results = []
+    for downloaded_csv in DOWNLOADED_FILES:
+        with open(downloaded_csv, 'r') as csv_file:
+            reader = csv.reader(csv_file)
+            parsed_header_row = False
+            desired_columns = []
+            for row in reader:
+
+                # Get desired columns if test_case_metrics were passed
+                if not parsed_header_row:
+                    if test_case_metrics is not None:
+                        for column_index, column_header in enumerate(row):
+                            if column_header in test_case_metrics:
+                                desired_columns.append(column_index)
+                    parsed_header_row = True
+
+                # Actually get the data
+                if test_case_ids is not None and row[0] in test_case_ids:
+                    query_results.append(_parse_row(test_case_metrics, row, desired_columns))
+                elif test_case_ids is None:
+                    query_results.append(_parse_row(test_case_metrics, row, desired_columns))
+                    print("row " + str(row))
+            csv_file.close()
+    return query_results
+
+
+def _parse_row(test_case_metrics, row, desired_columns):
+    """
+    Queries the desired information from the downloaded csvs
+    :param test_case_metrics: the desired test case metrics
+    :param row: the row being parsed
+    :param desired_columns: the columns to include
+    """
+    row_to_add = []
+    if test_case_metrics is not None:
+        for column_number in desired_columns:
+            row_to_add.append(row[int(column_number)])
+        pass
+    else:
+        row_to_add = row
+    return row_to_add
+
+
+if __name__ == '__main__':
+    # parse parameters
+    parser = argparse.ArgumentParser(prog='MetricsQueryTool', description='Queries the metrics.')
+    parser.add_argument('-b', '--branch', required=True)
+    parser.add_argument('-sw', '--startweek', required=False)
+    parser.add_argument('-ew', '--endweek', required=False)
```

Comment:
Are you able to add the "Past X weeks" arg like we discussed?

Reply:
No. The script has no realistic way of knowing what the current week is, unless the user passes it, which at that point we may as well have the user pass the start week and end week instead of current week + a number of weeks.

The script could also parse every file in the bucket to figure out what the current week is, but then we'd be at n squared for run time instead of n and I felt that was a bad compromise for a minor convenience.